### PR TITLE
Convert to .5 format

### PR DESCRIPTION
**Pull request by josmo**
Converting to the .5 format for beanstalk
Adding vendorized versions

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -1,33 +1,20 @@
...
-    email: $$DOCKER_EMAIL
-    repo: plugins/drone-elastic-beanstalk
-    tag: develop
-    when:
-      branch: develop
+pipeline:
+    build:
+      image: golang:1.6
+      commands:
+        - CGO_ENABLED=0 go build
```
Comment (tboerger) [inline .drone.yml]: You don't set any version

File: .drone.yml [hunk 2]
```diff
@@ -1,33 +1,20 @@
...
+    docker:
+      username: $$DOCKER_USER
```
Comment (tboerger) [inline .drone.yml]: The secrets system changed a lot, take a look at https://github.com/drone/drone-plugin-starter/blob/master/.drone.yml

File: main.go [hunk 3]
```diff
@@ -1,113 +1,104 @@
...
-			ApplicationName:       aws.String(vargs.Application),
-			Description:           aws.String(vargs.Description),
-			AutoCreateApplication: aws.Bool(vargs.AutoCreate),
-			Process:               aws.Bool(vargs.Process),
-			SourceBundle: &elasticbeanstalk.S3Location{
-				S3Bucket: aws.String(vargs.BucketName),
-				S3Key:    aws.String(vargs.BucketKey),
-			},
+	app := cli.NewApp()
+	app.Name = "s3 artifact plugin"
```
Comment (tboerger) [inline main.go]: `s3 artifact plugin`?

File: plugin.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+	if err != nil {
+		log.WithFields(log.Fields{
+			"error": err,
+		}).Error("Problem create application")
+		return err
+	}
+	if p.EnvironmentUpdate == true && err == nil {
+
+		if p.EnvironmentName == "" {
+			fmt.Println("Can't update environment without environment name")
```
Comment (tboerger) [inline plugin.go]: `return fmt.Errorf()` instead of `fmt.Println` and `os.Exit(1)`

Comment (tboerger) [inline plugin.go]: Or even logger like on the other places.

**Comment by josmo** [comment]
@tboerger everything ok for you? let me know if I need to change anything :)

**Comment by tboerger** [comment]
LGTM

**Comment by josmo** [comment]
@bradrydzewski If you're ok to merge this in. I can also take over this plugin and start publishing images.

**Pull request merged**